src/process_utility.c

```diff
@@ -3264,6 +3283,10 @@ process_altertable_end_subcmd(Hypertable *ht, Node *parsetree, ObjectAddress *ob
 			break;
 		case AT_AddColumn:
 		case AT_AddColumnRecurse:
+			Assert(IsA(cmd->def, ColumnDef));
+			if (TS_HYPERTABLE_HAS_COMPRESSION_TABLE(ht))
+				ts_cm_functions->process_altertable_cmd(ht, cmd);
```

Comment:
This cross-module call should not be necessary since we already forward DDL commands to TSL, which means you can capture it there instead of defining a new cross-module function. I don't believe it is a scalable approach to define new functions for every DDL and alter command.

Reply:
The earlier version of the function had subcmd specifc processing in src/process_utility.c. With this approach, I have  a more generic tsl side function for the ALTER TABLE ddl. This is invoked only when needed.

The downside of passing the entire processing over to ddl_command_start and ddl_command_end in the tsl code is: we would be replicating code for table lookup and subcmd processing on tsl side (see code path for process_altertable_end_table - has simple cmd and multiple cmds code path. If we were to do this in ddl_command_end, we  would replicate all this code,  but all this processing has already been done by the caller. I don't see this approach as simplifying the code) 

I can move the call at line 3288 to the end of the function process_altertable_end_subcmd to make it more generic. There is nothing that prevents the tsl  side function from being generic.